Include/internal/pycore_pymem_init.h

```diff
@@ -18,17 +18,31 @@ extern void * _PyMem_RawRealloc(void *, void *, size_t);
 extern void _PyMem_RawFree(void *, void *);
 #define PYRAW_ALLOC {NULL, _PyMem_RawMalloc, _PyMem_RawCalloc, _PyMem_RawRealloc, _PyMem_RawFree}
 
-#ifdef WITH_PYMALLOC
+#ifdef WITH_MIMALLOC
+extern void* _PyMem_MiMalloc(void *ctx, size_t size);
+extern void* _PyMem_MiCalloc(void *ctx, size_t nelem, size_t elsize);
+extern void* _PyMem_MiRealloc(void *ctx, void *ptr, size_t size);
+extern void _PyMem_MiFree(void *ctx, void *ptr);
+extern void* _PyObject_MiMalloc(void *ctx, size_t nbytes);
+extern void* _PyObject_MiCalloc(void *ctx, size_t nelem, size_t elsize);
+extern void* _PyObject_MiRealloc(void *ctx, void *ptr, size_t nbytes);
+extern void _PyObject_MiFree(void *ctx, void *ptr);
+
+#  define PYMEM_ALLOC {NULL, _PyMem_MiMalloc, _PyMem_MiCalloc, _PyMem_MiRealloc, _PyMem_MiFree}
+#  define PYOBJ_ALLOC {NULL, _PyObject_MiMalloc, _PyObject_MiCalloc, _PyObject_MiRealloc, _PyObject_MiFree}
```

Comment:
I would prefer to no use mimalloc by default for now. Not in this PR.

Maybe you can write a follow PR to use it by default if Python is built by NoGIL?